app/src/main/kotlin/com/vestrel00/daggerbutterknifemvp/ui/common/presenter/BasePresenter.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018 Vandolf Estrellado
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.vestrel00.daggerbutterknifemvp.ui.common.presenter
+
+import com.vestrel00.daggerbutterknifemvp.ui.common.view.MVPView
+
+/**
+ * Abstract [Presenter] for all presenters to extend.
+ *
+ * @param [T] the type of the [MVPView].
+ */
+abstract class BasePresenter<out T : MVPView>
+protected constructor(protected val view: T) : Presenter
```

Comment:
`Presenter` methods need not be implemented. See https://github.com/vestrel00/android-dagger-butterknife-mvp/pull/67/files#r164869368